includes/admin/extensions/class-extension-manager.php

```diff
@@ -0,0 +1,520 @@
+							?>
+							<div class="edd-extension-manager__step">
+								<?php $this->button( $inactive_parameters ); ?>
+							</div>
+							<?php
+						}
+						?>
+						<div class="edd-extension-manager__step">
+							<?php $this->link( $active_parameters ); ?>
+						</div>
+					</div>
+				</div>
+			</div>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Gets the extension card data.
+	 * Generally, the data from the products API will be used,
+	 * but extending classes can override that in the configuration array.
+	 *
+	 * @since 2.11.x
+	 * @param object $product The extension data from the Products API.
+	 * @return array
+	 */
+	private function get_card_data( $product ) {
+		$data = array();
+		if ( ! empty( $product['custom_title'] ) ) {
+			$data['title'] = $product['custom_title'];
+		} elseif ( ! empty( $product['title'] ) ) {
+			/* translators: the name of the extension */
+			$data['title'] = sprintf( __( 'Get %s Today!', 'easy-digital-downloads' ), $product['title'] );
+		}
+		if ( ! empty( $product['custom_description'] ) ) {
+			$data['description'] = $product['custom_description'];
+		}
+
+		return array_merge( $product, $data );
+	}
+
+	/**
+	 * Outputs the button to activate/install a plugin/extension.
+	 * If a link is passed in the args, create a button style link instead (@uses $this->link()).
+	 *
+	 * @since 2.11.x
+	 * @param array $args The array of parameters for the button.
+	 * @return void
+	 */
+	public function button( $args ) {
+		if ( ! empty( $args['href'] ) ) {
+			$this->link( $args );
+			return;
+		}
+		$defaults = array(
+			'button_class' => 'button-primary',
+			'plugin'       => '',
+			'action'       => '',
+			'button_text'  => '',
+			'type'         => 'plugin',
+			'id'           => '',
+			'product'      => '',
+			'pass'         => $this->required_pass_id,
+		);
+		$args     = wp_parse_args( $args, $defaults );
+		if ( empty( $args['button_text'] ) ) {
+			return;
+		}
+		?>
+		<button
+			class="button <?php echo esc_attr( $args['button_class'] ); ?> edd-extension-manager__action"
+			<?php
+			foreach ( $args as $key => $attribute ) {
+				if ( empty( $attribute ) || in_array( $key, array( 'button_class', 'button_text' ), true ) ) {
+					continue;
+				}
+				printf(
+					' data-%s="%s"',
+					esc_attr( $key ),
+					esc_attr( $attribute )
+				);
+			}
+			?>
+		>
+			<?php echo esc_html( $args['button_text'] ); ?>
+		</button>
+		<?php
+		add_action( 'admin_print_footer_scripts', function() {
+			if ( ! wp_script_is( 'edd-extension-manager', 'enqueued' ) ) {
+				wp_print_scripts( 'edd-extension-manager' );
+			}
+		} );
+	}
+
+	/**
+	 * Gets the download URL.
+	 *
+	 * @param string|int $url_or_item_id  Either the download URL (for a plugin) or the item ID (for an extension).
+	 * @param string     $type            Plugin or extension.
+	 * @return string|false Returns the download URL if possible, or false if not.
+	 */
+	public function get_download_url( $url_or_item_id, $type = 'plugin' ) {
```

Comment:
I'm finding this method a little odd, since it just returns exactly what was passed in (provided the `$type === 'plugin'`). Can you explain why you find it useful?

Reply:
If the type was not a plugin, the first parameter would have been a product ID and this method called up a class to get the download package from the licensing API. I left the method in as hopefully a future placeholder but I can remove it.

Reply:
Based on the context, I feel like right now it's acting more like `can_product_be_downloaded()` (given that if it returns `false`, you then do `wp_send_json_error()`).

I think for clarity it would be better to either remove it or rename it. Once we work on the licensing API stuff we can add it back in as it was if necessary.